### ISSUE TITLE
luci-app-firewall: Mixing zone- and attribute based rules in 'Traffic Rules' doesn't work as expected

### ISSUE THREAD

**Issue by manuel-kraus-19** [openwrt/luci#5299]
Environment: 19.07.7, package luci-app-firewall is git-21.222.69112-b41f377-1

Description:

The idea was to allow HTTP and other useful protocols for a bunch of zones using dedicated 'accept' entries in 'Traffic Rules', while having all zone defaults to 'forward reject' in 'General Settings'. A simple whitelist approach allowing only wanted protocols for most of the zones I have. Now I decided, that only one of my of zones shouldn't have any internet access at all. So I've put a source zone reject entry to the top of 'Traffic Rules' and it didn't work! That zone had still access to all the allowed protocols defined further down.. I got aware of that quite too late, weeks after setup.

After looking into that, my observation is: Zone-only based rules are being linked within the zone forward tables, which are below the attribute (IP, protocol, port) based rules in the FORWARD table.

So far it's not possible to have zone-only based rules taking precedence over attribute based rules using Luci (intentionally ignoring 'Custom Rules' here). There is no warning to the (casual) user and having those sliders for reordering rules is misguiding in this scenario, because there's no effect.

I have no access to any 20.x. device, so pardon me, if this issue is fixed there.

**Comment by jow-** [comment]
This is a result of the implementation design of the firewall3 backend, nothing that can be fixed in LuCI.

**Comment by manuel-kraus-19** [comment]
Okay, but we shouldn't drop that one too fast. Unfortunately the given implementation design of firewall3 becomes a dangerous flaw, if managed by LuCI at this point. It should be WYSIWYG by any means. That is what users expect. It's very important to prevent users from making the wrong assumptions, as reported in the case I made. I understand, that the implementation design of firewall3 is the root cause here, but you might be able to help at the presentation level of LuCI.
What about displaying the two classes of rules in two different sub sections of 'Traffic Rules'? The attribute based in the first, the zone-only in the second. A kind of visual separation (and having the GUI restricting the movement of rules between them) would make it clear, that there's no way (and sense at the end) to have them mixed. People would start to ask themselves the right questions and it might reduce the chance having people fall for the same trap, as I did.
